dpx/src/dpx_pdfobj.rs

```diff
@@ -988,8 +991,9 @@ pub unsafe fn pdf_copy_name(name: *const i8) -> *mut pdf_obj {
     let data = new(::std::mem::size_of::<pdf_name>() as u32) as *mut pdf_name;
     (*result).data = data as *mut libc::c_void;
     let length = strlen(name);
-    let name = CStr::from_ptr(name).to_owned();
     if length != 0 {
+        let slice = std::slice::from_raw_parts(name as *const u8, length as _);
+        let name = CString::new(slice).unwrap();
```

Comment:
pdf_copy_name is not in the C source, so I'm not sure what it's intended to do. Previously, it was not copying anything, which seemed weird for a function called 'copy'.

Reply:
I suspect this would be a problem on drop, when multiple CStr::into_raw 'shallow copies' would each be converted back to a CString.